src/executables/sysrepo-plugind.c

```diff
@@ -527,6 +534,34 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    /* No need to create a pid file if we're in debug mode (we're not a daemon) */
+
+    if (debug && pidfile) {
+        SRPLG_LOG_WRN("sysrepo-plugind", "Ignoring pidfile in debug mode.");
+    }
```

Comment:
This seems redundant to me. Users specify all the parameters and if they use both `-d` and `-p` I would assume it is what they wanted to do.

Reply:
The idea was that if you're running it in debug mode, it's most likely not running in the context of `init` or some other daemon orchestrating process. You know the pid in debug mode. I have however removed it per request.

Reply:
I do not think trying to guess what the users wants to do is a game you can win, it is best to simply do what the user requests.